src/screens/scanning/ManualVoucherScreen.tsx

```diff
@@ -30,6 +34,51 @@ function ManualVoucherScreen() {
   //   { label: 'Banana', value: 'banana' },
   // ]);
 
+  const SNonChange = (text: string) => {
+    const value = text.replace(/\D/g, '');
+    setID(value);
+  };
+
+  const VAonChange = (text: string) => {
+    const value = text.replace(/[^\d.]/g, '');
+    setVoucherAmount(value);
+  };
+
+  const validateSerialNumberInput = () => {
+    try {
+      const SNSchema = z.coerce.number().gt(0);
+      SNSchema.parse(transactionID);
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.log(error);
+    }
+  };
+
+  const validateCurrencyFormat = (input: string) => {
```

Comment:
I would move these validation functions to a separate util file like `validationUtils.ts` or `voucherUtils.ts` in the `src/utils` folder.